python/hail/dataset.py

```diff
@@ -915,6 +915,256 @@ def annotate_variants_vds(self, other, expr=None, root=None):
+                                "regulatory_region_variant"
+                            else if (csq_terms.exists(t => t.contains("feature_truncation"))) 
+                                "feature_truncation"
+                            else if (csq_terms.exists(t => t.contains("intergenic_variant"))) 
+                                "intergenic_variant"
+                            else
+                                va.vep.most_severe_consequence
+                        """
+                    )
+                    .annotate_variants_expr(
+                        """
+                        va.gene.gene_symbol = 
+                            let canonical_transcripts = va.vep.transcript_consequences.filter(t => t.canonical == 1) in
+                            if (canonical_transcripts.exists(t => t.consequence_terms.toSet.contains(va.gene.consequence)))
+                                canonical_transcripts.find(t => t.consequence_terms.toSet.contains(va.gene.consequence)).gene_symbol
+                            else
+                                va.vep.transcript_consequences.find(t => t.consequence_terms.toSet.contains(va.gene.consequence)).gene_symbol,
+                        va.gene.gene_id =
+                            let canonical_transcripts = va.vep.transcript_consequences.filter(t => t.canonical == 1) in
+                            if (canonical_transcripts.exists(t => t.consequence_terms.toSet.contains(va.gene.consequence)))
+                                canonical_transcripts.find(t => t.consequence_terms.toSet.contains(va.gene.consequence)).gene_id
+                            else
+                                va.vep.transcript_consequences.find(t => t.consequence_terms.toSet.contains(va.gene.consequence)).gene_symbol,
+                        va.transcript_id = let canonical_transcripts = va.vep.transcript_consequences.filter(t => t.canonical == 1) in
+                            if (canonical_transcripts.exists(t => t.consequence_terms.toSet.contains(va.gene.consequence)))
+                                canonical_transcripts.find(t => t.consequence_terms.toSet.contains(va.gene.consequence)).transcript_id
+                            else
+                                va.vep.transcript_consequences.find(t => t.consequence_terms.toSet.contains(va.gene.consequence)).transcript_id
+                        """
+                    )
+                )
+
+            # drop VEP annotations if not specifically requested
+            if not veps:
+                self = self.annotate_variants_expr('va = drop(va, vep)')
+
+            # subset VEP annotations if needed
+            subset = ','.join([x.rsplit('.')[-1] for x in annotations if x.startswith('va.vep.')])
+            if subset:
+                self = self.annotate_variants_expr('va.vep = select(va.vep, {})'.format(subset))
+
+        # iterate through files, selected annotations from each file
+        for file, expr in file_exprs.iteritems():
```

Comment:
change the name 'file': it's good practice to avoid collisions with python standard functions